service/dashboard.js

```diff
@@ -0,0 +1,120 @@
+const {getCurrentWindow, globalShortcut} = require('electron').remote;
+let fs = require('fs');
+const electron = require('electron');
+const remote = electron.remote;
+const url = require('url');
+const path = require('path');
+//Reload function to reload page by pressing F5 or Control+R.
+//The purpose is for testing and to quickly show changes to HTML and
+//CSS
+var reload = ()=>{
+    getCurrentWindow().reload()
+}
+
+globalShortcut.register('F5', reload);
+globalShortcut.register('CommandOrControl+R', reload);
+
+window.addEventListener('beforeunload', ()=>{
+    globalShortcut.unregister('F5', reload);
+    globalShortcut.unregister('CommandOrControl+R', reload);
+})
+
+
+function searchForSongs() {
+    let searchType = null;
+
+    if (document.getElementById('search_song_title').checked == true) {
+        searchType = document.getElementById('search_song_title').value;
+    }
+    if (document.getElementById('search_song_artist').checked == true) {
+        searchType = document.getElementById('search_song_artist').value;
+    }
+    if (document.getElementById('search_song_release').checked == true) {
+        searchType = document.getElementById('search_song_release').value;
+    }
+    let searchField = document.getElementById('search_song').value;
+
+    //console.log("search Type: " + searchType + "What's searched: " + searchField);
+    if(searchType == null || searchField == '' || searchField == null) {
+        alert('Please fill in all required fields');
+    } else {
+        fs.readFile(__dirname + '/../data/music.json', (err, rawData) => {
+            if(err) {
+                console.log(err);
+            } else {
+                let data = JSON.parse(rawData);
+                let list = document.querySelector('ul');
+                list.innerHTML = '';
+                let id = 'song_';
+                let idNum = 1;
+                data.forEach(element => {
```

Comment:
I thought angel was going to implement the search service...did you talk to him about that? Also not sure if the professor wanted the option to be able to chose which category to search through; Artist, Song, Release. Will have to get that checked if we decide to use this algorithm.

Reply:
I talked to him and I'm going to go ahead and use his code.  But I'll go ahead and push this now so that once he's pushed his I'll change my function by integrating it with his.